LIPS/lip-31.md

```diff
@@ -0,0 +1,630 @@
+- **Minimal coordination**: Adding a new vault type requires only deploying the vault contract and configuring it in the registry — likely no changes to Lido Core, stETH token, or existing integrations.
+- **Future-proof**: The design accommodates potential future developments like distributed validators, new consensus mechanisms, or cross-chain staking without architectural changes.
+
+### Minimal Lido Core surface area
+
+The design carefully limits changes to the battle-tested Lido Core:
+
+- **Contained complexity**: New complexity is isolated in VaultHub and vault implementations, while Lido Core adds only essential external share accounting functions.
+- **Preserved invariants**: Core protocol invariants around deposits, withdrawals, and rebasing remain unchanged; external shares are handled as a parallel accounting system.
+- **Agnostic to vault logic**: Lido Core doesn't need to understand vault-specific logic, strategies, or operational details—it only enforces the global backing invariant.
+
+## Backward compatibility
+
+The implementation of this LIP prioritizes maintaining full backward compatibility with existing Lido infrastructure and integrations. The transition to support external ether sources is designed to be seamless for current users and integrators:
+
+### Existing deposit flow preservation
+
+The traditional ETH staking flow through Lido Core remains completely unchanged:
+
+- **Direct deposits**: Users can continue depositing ETH directly to the Lido contract, receiving stETH 1:1 as before. The deposit flow through `submit()` and referral functions operates identically.
+- **Deposit router**: The existing deposit distribution logic and staking module interfaces remain intact. Current staking modules continue operating without modification.
+- **Buffer management**: The buffered ether mechanics, withdrawal request handling, and deposit allocation algorithms function as they do today.
+- **Validator lifecycle**: The process of spinning up validators, managing exits, and handling rewards through the existing Node Operator Registry remains unchanged for Lido Core.
+
+### wstETH wrapper compatibility
+
+The wrapped stETH (wstETH) contract and its deployments across L1 and L2s maintain full compatibility:
+
+- **Unchanged interface**: The wstETH wrapping/unwrapping mechanics remain identical, as they depend only on stETH's share-based accounting, which is preserved.
+- **Cross-chain bridges**: Existing bridge implementations for wstETH on Arbitrum, Base, Optimism, Polygon, and other L2s continue functioning without modification.
+- **Share rate consistency**: The fundamental share rate calculation preserves the wstETH:stETH exchange rate continuity, ensuring no disruption to existing positions.
+- **Integration stability**: DeFi protocols using wstETH as collateral or for liquidity provision experience no changes in behavior or required integration updates.
+
+### Withdrawal system compatibility
```

Comment:
The actors between whom the discount assigned to finalized applications in WQ is socialized are changing. Now, minted fees to operators and treasury are excluded from those who receive these surpluses.

Reply:
gud